homeassistant/components/surepetcare/sensor.py

```diff
@@ -0,0 +1,130 @@
+"""Support for Sure PetCare Flaps/Pets sensors."""
+import logging
+
+from homeassistant.const import (
+    ATTR_VOLTAGE,
+    CONF_ID,
+    CONF_NAME,
+    CONF_TYPE,
+    DEVICE_CLASS_BATTERY,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+
+from .const import (
+    DATA_SURE_PETCARE,
+    SPC,
+    SURE_BATT_VOLTAGE_DIFF,
+    SURE_BATT_VOLTAGE_LOW,
+    TOPIC_UPDATE,
+    SureThingID,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up Sure PetCare Flaps sensors."""
+    spc = hass.data[DATA_SURE_PETCARE][SPC]
+    async_add_entities(
+        [
+            FlapBattery(thing[CONF_ID], thing[CONF_NAME], spc)
+            for thing in spc.ids
+            if thing[CONF_TYPE] == SureThingID.FLAP.name
+        ]
+    )
+
+
+class FlapBattery(Entity):
+    """Sure Petcare Flap."""
+
+    def __init__(self, _id: int, name: str, spc, data: dict = None):
+        """Initialize a Sure Petcare Flap battery sensor."""
+        self._id = _id
+        self._name = f"Flap {name.capitalize()} Battery Level"
+        self._spc = spc
+        self._unit_of_measurement = "%"
+        self._state = self._spc.states[SureThingID.FLAP.name][self._id].get("data")
+
+    @property
+    def should_poll(self):
+        """Return true."""
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return battery level in percent."""
+        try:
+            per_battery_voltage = self._state["battery"] / 4
+            voltage_diff = per_battery_voltage - SURE_BATT_VOLTAGE_LOW
+            battery_percent = int(voltage_diff / SURE_BATT_VOLTAGE_DIFF * 100)
+        except (KeyError, TypeError):
+            battery_percent = None
+
+        return battery_percent
+
+    @property
+    def unique_id(self):
+        """Return an unique ID."""
+        return f"{self._spc.household_id}-{self._id}"
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        return DEVICE_CLASS_BATTERY
+
+    @property
+    def device_state_attributes(self):
+        """Return state attributes."""
+        attributes = None
+        if self._state:
+            try:
+                voltage_per_battery = float(self._state["battery"]) / 4
+                attributes = {
+                    ATTR_VOLTAGE: f"{float(self._state['battery']):.2f} V",
```

Comment:
Don't add unit at the end of the value. We want state attributes easily usable in automations. Units can be documented in the integration docs.